YoutubeDownloader/Views/Components/DashboardView.xaml.cs

```diff
@@ -1,23 +1,41 @@
-﻿using System.Windows.Input;
+using Avalonia.Controls;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using YoutubeDownloader.ViewModels.Components;
+using YoutubeDownloader.Views.Framework;
 
 namespace YoutubeDownloader.Views.Components;
 
-public partial class DashboardView
+public partial class DashboardView : ViewModelAwareUserControl<DashboardViewModel>
 {
     public DashboardView()
     {
         InitializeComponent();
+        QueryTextBox.AddHandler(
+            InputElement.KeyDownEvent,
+            (sender, args) =>
+            {
+                // Disable new lines when pressing enter without shift
+                if (args.Key == Key.Enter && args.KeyModifiers != KeyModifiers.Shift)
+                {
+                    args.Handled = true;
+
+                    // We handle the event here so we have to directly "press" the default button
+                    ProcessQueryButton.Command?.Execute(ProcessQueryButton.CommandParameter);
+                }
+            },
+            RoutingStrategies.Tunnel
+        );
     }
 
-    private void QueryTextBox_OnPreviewKeyDown(object sender, KeyEventArgs args)
+    public async void OnStatusPointerReleased(object sender, PointerReleasedEventArgs args)
     {
-        // Disable new lines when pressing enter without shift
-        if (args.Key == Key.Enter && args.KeyboardDevice.Modifiers != ModifierKeys.Shift)
+        if (
+            args.Pointer.IsPrimary
+            && sender is TextBlock { DataContext: DownloadViewModel downloadViewModel }
+        )
         {
-            args.Handled = true;
-
-            // We handle the event here so we have to directly "press" the default button
-            AccessKeyManager.ProcessKey(null, "\x000D", false);
+            await downloadViewModel.CopyErrorMessage();
```

Comment:
Probably can just do `DataContext.CopyErrorMessage` now and simplify the `if`

Reply:
You mean because of the ViewModelAwareUserControl generics? That won't work, because the DataContext of the view is DashboardViewModel. Or do you mean just doing casting on the call site instead of type check in the if?